### ISSUE TITLE
[Accessibility]: Code Insights Creation UI - Clicking done button doesn't announce anything

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41669]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33494

### Problem description

When clicking the "Done" button on a data series, it collapses the panel but doesn't announce that anything has changed.

This leads the user to believe nothing happened.

### Expected behavior

Should announce that the panel has collapsed and inform the user of the current focus.

### Additional details

Note this is related to https://github.com/sourcegraph/sourcegraph/issues/41664, which deals with the lack of focus. That focus problem should probably be addressed first.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.
